src/Form/DatasetLinkType.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace App\Form;
+
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+use App\Entity\DatasetLink;
+
+/**
+ * A form for creating Dataset Links.
+ */
+class DatasetLinkType extends AbstractType
+{
+    /**
+     * Builds the form.
+     *
+     * @param FormBuilderInterface $builder The form builder.
+     * @param array                $options The options.
+     *
+     * @see FormTypeExtensionInterface::buildForm()
+     *
+     * @return void
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $builder
+            ->add('url', Type\TextType::class, array(
+                'label' => 'Link Url:',
+                'required' => false,
+            ))
+            ->add('name', Type\TextType::class, array(
```

Comment:
We can try using Datalist here, and the ticket describes this to be a predefined list.